packages/electron-updater/src/differentialDownloader/DifferentialDownloader.ts

```diff
@@ -233,7 +233,7 @@ export abstract class DifferentialDownloader {
 
         const request = this.httpExecutor.createRequest(requestOptions, response => {
           response.on("error", reject)
-          response.on("abort", () => {
+          response.on("aborted", () => {
```

Comment:
So it looks like `response` is an `IncomingMessage` object (not sure why it is typed as any), but with it typed correctly the only events that can be listened to according to intellisense are:
<img width="574" alt="Screenshot 2024-06-29 at 8 40 14 AM" src="https://github.com/electron-userland/electron-builder/assets/5238867/8817d1ad-7954-490c-844d-80f3fcce9e62">

I'm wondering if this implementation is legacy for older versions of nodejs?
Note: I think older versions of nodejs still need to be supported though otherwise a bunch of electron versions that electron-updater supports would need to be dropped.
Node 14 - https://nodejs.org/docs/latest-v14.x/api/http.html#http_event_abort
Node 16 - https://nodejs.org/docs/latest-v16.x/api/http.html#event-abort
> Stability: 0 - Deprecated. Listen for the 'close' event instead.

So it looks like `abort` is indeed correct, but for newer versions of nodejs, we might need to leverage something akin to:
```
response.on('close', () => {
   const isAborted = response.aborted
   // other logic
})
```

Reply:
From the Node.js type definitions, not only are there these events, but you can also customize other events. 
![image](https://github.com/user-attachments/assets/01d64460-b039-4907-82c8-0bb383c9297d)

Looking at the response definitions, you can see there is an `aborted` event but no `abort` event.
![image](https://github.com/user-attachments/assets/406d244d-ea87-4fb6-9703-ec2103e93377)

Reply:
Based on the official Node.js issues and the full download code(7 years ago) from electron-builder, it seems that the `aborted` event is correct.

https://github.com/nodejs/node/issues/31630

https://github.com/electron-userland/electron-builder/blob/1c4ec128ae877ad3a3a3c2765e98e59329fd900d/packages/builder-util-runtime/src/httpExecutor.ts#L140

Reply:
Nice investigation work! I'm in favor of proceeding with this PR change then